### PR TITLE
microsoft_exchange_online_message_trace: map fields to ECS

### LINKED ISSUE #5083: [Microsoft Exchange Online Message Trace] Missing `event` ECS field index mappings
The Microsoft Exchange Online Message Trace integration only maps `event.end`, but there are a number of `event` fields that the integration sets, but doesn't map.

It is at least missing:
- event.dataset
- event.outcome

### PR DESCRIPTION
**Pull request by efd6**
Also retain original fields in data stream object.

## What does this PR do?

This improves the ECS compliance of the package.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Closes #5083

### CODE REVIEW

File: packages/microsoft_exchange_online_message_trace/data_stream/log/_dev/test/pipeline/test-log.log-expected.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "expected": [
         {
-            "@timestamp": "2022-09-05T18:10:13.4907658",
+            "@timestamp": "2022-09-05T18:10:13.490Z",
```
Comment (kcreddy) [inline packages/microsoft_exchange_online_message_trace/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:4]: In this case, could we use [date_nanos](https://www.elastic.co/guide/en/elasticsearch/reference/current/date_nanos.html) datatype to support precision?

File: packages/microsoft_exchange_online_message_trace/data_stream/log/_dev/test/pipeline/test-log.log-expected.json [hunk 2]
```diff
@@ -47,7 +47,182 @@
...
+                "online_message_trace": {
+                    "EndDate": "2022-09-05T21:46:46.4206759Z",
+                    "FromIP": "81.2.69.144",
+                    "Index": 0,
+                    "MessageId": "\u003ca210cf91-4f2e-484c-8ada-3b27064ee5e3@az.uksouth.production.microsoft.com\u003e",
+                    "MessageTraceId": "cf7a249a-5edd-4350-130a-08da8f69e0f6",
+                    "Organization": "contoso.com",
+                    "Received": "2022-09-05T18:10:13.4907658",
+                    "RecipientAddress": "linus@contoso.com",
+                    "SenderAddress": "azure-noreply@azure.microsoft.com",
```
Comment (kcreddy) [inline packages/microsoft_exchange_online_message_trace/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:63]: I think we could also extract `azure-noreply` as `source.user.name` or `source.user.id`

File: packages/microsoft_exchange_online_message_trace/data_stream/log/_dev/test/pipeline/test-log.log-expected.json [hunk 3]
```diff
@@ -47,7 +47,182 @@
...
+            "microsoft": {
+                "online_message_trace": {
+                    "EndDate": "2022-09-05T21:46:46.4206759Z",
+                    "FromIP": "81.2.69.144",
+                    "Index": 0,
+                    "MessageId": "\u003ca210cf91-4f2e-484c-8ada-3b27064ee5e3@az.uksouth.production.microsoft.com\u003e",
+                    "MessageTraceId": "cf7a249a-5edd-4350-130a-08da8f69e0f6",
+                    "Organization": "contoso.com",
+                    "Received": "2022-09-05T18:10:13.4907658",
+                    "RecipientAddress": "linus@contoso.com",
```
Comment (kcreddy) [inline packages/microsoft_exchange_online_message_trace/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:62]: In the [ECS doc here,](https://www.elastic.co/guide/en/ecs/current/ecs-user-usage.html#:~:text=When%20a%20system%20uses%20an%20email%20address%20as%20the%20main%20identifier%2C%20populate%20both%20user.id%20and%20user.email%20with%20it.) it says that if the user info is provided as an email address, we should be using `user.id` and `user.email` fields instead.

File: packages/microsoft_exchange_online_message_trace/data_stream/log/_dev/test/pipeline/test-log.log-expected.json (hunk 1 above)
Comment (efd6) [reply packages/microsoft_exchange_online_message_trace/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:4]: I've changed the package's own fields to use this type. I don't see it used anywhere else in integrations, so I'm doubtful about using it for `@timestamp`.

File: packages/microsoft_exchange_online_message_trace/data_stream/log/_dev/test/pipeline/test-log.log-expected.json (hunk 2 above)
Comment (efd6) [reply packages/microsoft_exchange_online_message_trace/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:69]: Done

File: packages/microsoft_exchange_online_message_trace/data_stream/log/_dev/test/pipeline/test-log.log-expected.json (hunk 3 above)
Comment (efd6) [reply packages/microsoft_exchange_online_message_trace/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:68]: I think I've covered this now.

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**